### PR TITLE
Runtime usage counters

### PR DESCRIPTION
**Pull request by Wiezzel**
Defined a set of common counters for Runtime objects. Implemented the counters for Docker environment.

**Do not merge. It's not supposed to go into 0.22 release.**

### CODE REVIEW

**Comment by Wiezzel** [comment]
@mfranciszkiewicz @maaktweluit Please review. It's a draft because I don't want to merge it to b0.22. Will make a proper PR once b0.22 is merged to develop.

File: golem/envs/docker/cpu.py [hunk 1]
```diff
@@ -233,6 +239,48 @@ def _update_status_loop(self) -> None:
...
+                self._counters.cpu_total_ns = max(
+                    self._counters.cpu_total_ns,
+                    cpu_stats['total_usage'])
+
+                mem_stats = stats['memory_stats']
+                logger.debug("RAM usage: %r", mem_stats)
+                self._counters.ram_max_bytes = mem_stats['max_usage']
+                total_usage = self._counters.ram_avg_bytes * self._num_samples
+                self._counters.ram_avg_bytes = (
+                    (total_usage + mem_stats['usage']) /
```
Comment (mfranciszkiewicz) [inline golem/envs/docker/cpu.py:283]: `0` values will still slightly skew the average but that shouldn't matter in the long run

File: golem/envs/docker/cpu.py [hunk 2]
```diff
@@ -233,6 +239,48 @@ def _update_status_loop(self) -> None:
         self._logger.info("Runtime is no longer running. "
                           "Stopping status update thread.")
 
+    def _update_counters(self) -> None:
+        start_time = time.time()
+        client = local_client()
+        stream = client.stats(self._container_id, decode=True, stream=True)
+        for stats in stream:
```
Comment (mfranciszkiewicz) [inline golem/envs/docker/cpu.py]: May throw `docker.errors.APIError` on server error. Maybe that should be handled as well?

**Review by mfranciszkiewicz — COMMENTED** [review]
Looks fine. One small doubt about the stability of `stream`

File: golem/envs/docker/cpu.py (hunk 1 above)
Comment (Wiezzel) [reply golem/envs/docker/cpu.py:283]: You mean the initial zeros? No, they won't skew anything. The expressions will evaluate as follows in the first turn of the loop:
```
total_usage = 0 * 0 = 0
self._counters.ram_avg_bytes = (0 + mem_stats['usage']) / (0 + 1) = mem_stats['usage']
```

Comment (mfranciszkiewicz) [reply golem/envs/docker/cpu.py:283]: I rather meant the increased `num_samples` and the zeros at the end. I think the formula is fine. Anyway, it's only relevant to (extremely) short-lived containers.

**Review by maaktweluit — APPROVED** [review]

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**